src/plone/testing/_z2_testbrowser.py

```diff
@@ -1,16 +1,15 @@
 # -*- coding: utf-8 -*-
 from cStringIO import StringIO
+from zope.testbrowser.testing import PublisherConnection
+from zope.testbrowser.testing import PublisherResponse
+
 import base64
 import mechanize
 import re
 import rfc822
 import sys
 import urllib
 import urllib2
-
-# Using a from-import here to avoid an AttributeError below when using
-# zope.testbrowser 4.x without zope.app.testing:
-from zope.testbrowser.testing import PublisherConnection, PublisherResponse
```

Comment:
`zope.testbrowser` 5 does neither have `PublisherConnection` nor `PublisherResponse` in [testing.py](https://github.com/zopefoundation/zope.testbrowser/blob/master/src/zope/testbrowser/testing.py). This means that the whole `_z2_testbrowser.py` module will not work with `zope.testbrowser` >= 5.0
The needed imports in `zope.testbrowser` were removed in [f01ac733b](https://github.com/zopefoundation/zope.testbrowser/commit/f01ac733b04036ff6bdbec944ca592a47a5e45c3#diff-69501ce4c6ec40f61f5fa8a2277baf4d) but the imported code still exists in [zope.app.testing](https://github.com/zopefoundation/zope.app.testing/blob/master/src/zope/app/testing/testbrowser.py) but it will not run with `zope.testbrowser` >= 5 as it still depends on `mechanize` which was removed in `zope.testbrowser` 5.0.

I am not sure who uses (aka can use) this module at all. (It requires to use `zope.app.testing` which IMHO is not that common in Zope 2 land and it can only work with an older `zope.testbrowser` version. But I do not know what plone does. I think the needed bits to run `zope.testbrowser` on Zope 2 or 4 have been moved to [Zope master](https://github.com/zopefoundation/Zope/blob/master/src/Testing/testbrowser.py) resp. [2.13 branch](https://github.com/zopefoundation/Zope/blob/2.13/src/Testing/testbrowser.py))

I'd suggest to remove this `_z2_testbrowser.py` module in another PR and release this as `plone.testing` 6.0.